### PR TITLE
Remove dependencies

### PR DESCRIPTION
**Pull request by antoyo**
I'll update this text with the compilation time for every change.
It turns out `git2-sys` is very slow to compile (+1 minute).
Maybe we can re-enable the features of `env_logger` since it does not help much.

## Before
Clean build: 4m 15s
Rebuild:     1m 30s

## After removing docopt
Clean build: 2m 26s
Rebuild:     1m 20s

## After disable features of env_logger
Clean build: 2m 23s
Rebuild:     1m 18s

## After removing features of regex
Clean build: 2m 11s
Rebuild:     1m 17s

## After removing git2
Clean build: 1m 28s
Rebuild:     55.24s

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -34,15 +61,18 @@ fn build_config() -> Result<Config, String> {
...
-        args.get_str("<library>"),
-        args.get_str("<version>"),
-        args.get_str("-o"),
-        args.get_str("--doc-target-path"),
-        args.get_bool("-b"),
-        args.get_bool("-s"),
+        matches.opt_str("d").as_ref().map(|string| string.as_str()),
+        matches.free.get(0).as_ref().map(|string| string.as_str()),
+        matches.free.get(1).as_ref().map(|string| string.as_str()),
+        matches.opt_str("o").as_ref().map(|string| string.as_str()),
```
Comment (EPashkin) [inline src/main.rs]: Maybe better add function for `.as_ref().map(|string| string.as_str())`?

**Comment by EPashkin** [comment]
@antoyo Thanks, faster is better.

@GuillaumeGomez, @sdroege What you think?

**Comment by GuillaumeGomez** [comment]
I'm the one who told him to do it. :stuck_out_tongue:

**Comment by EPashkin** [comment]
By way: as I remember rust don't use libgit2 but call git from command line, maybe we can do same.

File: src/main.rs [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
+    let program = args[0].clone();
+
+    let mut options = Options::new();
+    options.optopt(
+        "c",
+        "config",
+        "Config file path (default: Gir.toml)",
+        "CONFIG",
+    );
+    options.optflag("h", "help", "Show this message");
```
Comment (sdroege) [inline src/main.rs]: Compared to docopt or clap this is quite ugly :)

**Comment by sdroege** [comment]
>  It turns out `git2-sys` is very slow to compile (+1 minute).

I think it's much faster if you actually have libgit2 installed on your system :)

But yeah, works for me generally. I still think `getops` is rather ugly compared to alternatives but go for it if @EPashkin is fine with it too

**Comment by EPashkin** [comment]
@antoyo What is time with CLI git?

File: src/main.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/main.rs]: Yep but it's just options and quite a small amount of code so who cares. :)

**Comment by antoyo** [comment]
@EPashkin: I think it's a bit faster than with the previous attempt which was using `libgit2` as a shared library.

**Comment by antoyo** [comment]
This is ready to be merged.
The other crates to remove to improve compile time will require much more work: regex and toml.

**Comment by EPashkin** [comment]
@antoyo Thanks

**Pull request merged**